src/mngt/ocf_mngt_flush.c

```diff
@@ -142,55 +142,61 @@ static int _ocf_mngt_get_sectors(ocf_cache_t cache, ocf_core_id_t core_id,
 	ocf_core_id_t i_core_id;
 	struct flush_data *p;
 	uint32_t i, j, dirty = 0;
+	uint64_t step = 0;
+	unsigned ret = 0;
+
+	ocf_metadata_start_exclusive_access(&cache->metadata.lock);
 
 	dirty = env_atomic_read(&core->runtime_meta->dirty_clines);
 	if (!dirty) {
 		*num = 0;
 		*tbl = NULL;
-		return 0;
+		goto unlock;
 	}
 
 	p = env_vmalloc(dirty * sizeof(**tbl));
-	if (!p)
-		return -OCF_ERR_NO_MEM;
+	if (!p) {
+		ret = -OCF_ERR_NO_MEM;
+		goto unlock;
+	}
 
 	for (i = 0, j = 0; i < cache->device->collision_table_entries; i++) {
 		ocf_metadata_get_core_info(cache, i, &i_core_id, &core_line);
 
-		if (i_core_id != core_id)
-			continue;
-
-		if (!metadata_test_valid_any(cache, i))
-			continue;
-
-		if (!metadata_test_dirty(cache, i))
-			continue;
-
-		if (ocf_cache_line_is_used(cache, i))
-			continue;
+		if (i_core_id == core_id &&
+				metadata_test_valid_any(cache, i) &&
+				metadata_test_dirty(cache, i)) {
+			/* It's valid and dirty target core cacheline */
+			p[j].cache_line = i;
+			p[j].core_line = core_line;
+			p[j].core_id = i_core_id;
+			j++;
+
+			/* stop if all cachelines were found */
+			if (j == dirty)
+				break;
+		}
 
-		/* It's core_id cacheline and it's valid and it's dirty! */
-		p[j].cache_line = i;
-		p[j].core_line = core_line;
-		p[j].core_id = i_core_id;
-		j++;
-		/* stop if all cachelines were found */
-		if (j == dirty)
-			break;
+		if (++step % 100000 == 0) {
```

Comment:
I know it's a very miniscule optimization, but its cost is miniscule as well - could you make it a power of two?

Reply:
Also, is separate `step` variable really needed? It seems to be exact duplicate od `i`

Reply:
All done